### PR TITLE
Use module debug functions in more packages

### PR DESCRIPTION
**Pull request by mmun**
Part of #12204.

### CODE REVIEW

File: packages/ember-metal/lib/deprecate_property.js [hunk 1]
```diff
@@ -21,10 +21,6 @@ import { set } from 'ember-metal/property_set';
 */
 
 export function deprecateProperty(object, deprecatedKey, newKey, options) {
-  function deprecate() {
```
Comment (rwjblue) [inline packages/ember-metal/lib/deprecate_property.js]: This is also used by the `deprecateProperty` setter below (they share the same deprecation call).  The setter below now calls the imported `deprecate()`, but doesn't actually work..

**Comment by rwjblue** [comment]
:+1: - Looks good once the that one comment is addressed.

File: packages/ember-metal/lib/deprecate_property.js (hunk 1 above)
Comment (mmun) [inline packages/ember-metal/lib/deprecate_property.js]: Ack, good eye.

Comment (mmun) [inline packages/ember-metal/lib/deprecate_property.js]: This is fixed now.

**Pull request merged**